native/src/main/java/io/ballerina/stdlib/data/jsondata/json/JsonCreator.java

```diff
@@ -99,21 +99,25 @@ static Optional<BMap<BString, Object>> initNewMapValue(JsonParser.StateMachine s
                 RecordType recordType = (RecordType) currentType;
                 nextMapValue = ValueCreator.createRecordValue(expType.getPackage(), expType.getName());
                 sm.fieldHierarchy.push(new HashMap<>(recordType.getFields()));
```

Comment:
Modifications to the StateMachine seems to have a repeated pattern. Would it be possible to factor it out to a separate method that is parametrised with the restType and state machine?

Reply:
Updated [aef3dc8](https://github.com/ballerina-platform/module-ballerina-data.jsondata/pull/2/commits/aef3dc8abdd5f918b9b14ae1036f935be8739e64)